hnn_core/optimization/metrics.py

```diff
@@ -0,0 +1,72 @@
+"""Metrics for parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from hnn_core import simulate_dipole
+
+from scipy.signal import resample
+
+
+def _rmse_evoked(net, initial_params, set_params, predicted_params,
+                 update_params, obj_values, scale_factor, smooth_window_len,
+                 target, tstop):
+    """The objective function for evoked responses.
+
+    Parameters
+    ----------
+    net : Network
+        The network object.
+    initial_params : dict
+        Keys are parameter names, values are initial parameters.
+    set_params : func
+        User-defined function that sets network drives and parameters.
+    predicted_params : list
+        Parameters selected by the optimizer.
+    update_params : func
+        Function to update params.
+    scale_factor : float
+        The dipole scale factor.
+    smooth_window_len : float
+        The smooth window length.
+    tstop : float
+        The simulated dipole's duration.
+    target : ndarray
+        The recorded dipole.
+
+    Returns
+    -------
+    obj : float
+        Normalized RMSE between recorded and simulated dipole.
+    """
+
+    params = update_params(initial_params, predicted_params)
+
+    # simulate dpl with predicted params
+    new_net = net.copy()
+    set_params(new_net, params)
+    dpl = simulate_dipole(new_net, tstop=tstop, n_trials=1)[0]
```

Comment:
@rythorpe @carolinafernandezp  a really cheap way to speed this up will be adding the ability to set `dt`. `dt=0.5` is really the sweet spot of being embarrassingly large but still allows for proper testing.

Reply:
If we allow the constructor to accept arbitrary callables, maybe we could expose the `dt` parameter within `_dipole_rmse()` so that we can use it in the test as follows:

```python
from functools import partial

obj_fun = partial(_dipole_rmse, dt=0.5)
optimizer = Optimizer(...obj_fun=obj_func)
```